src/mlpack/methods/decision_tree/decision_tree_regressor_impl.hpp

```diff
@@ -663,7 +663,7 @@ double DecisionTreeRegressor<FitnessFunction,
       bestGain = dimGain;
 
       // If the gain is the best possible, no need to keep looking.
-      if (bestGain >= 0.0)
+      if (FitnessFunction::bestGainIsZero && bestGain >= 0.0)
```

Comment:
We could add this check to the classification decision tree too.  It might be useful just for uniformity between regression and classification loss functions. :+1:

Reply:
Agreed. I will do that 👍